Client/mods/deathmatch/logic/lua/CLuaFunctionParseHelpers.cpp

```diff
@@ -564,21 +564,27 @@ ADD_ENUM(D3DPT_TRIANGLESTRIP, "trianglestrip")
 ADD_ENUM(D3DPT_TRIANGLEFAN, "trianglefan")
 IMPLEMENT_ENUM_END("primitive-type")
 
-IMPLEMENT_ENUM_BEGIN(eVehicleDummies)
-ADD_ENUM(LIGHT_FRONT_MAIN, "light_front_main")
-ADD_ENUM(LIGHT_REAR_MAIN, "light_rear_main")
-ADD_ENUM(LIGHT_FRONT_SECONDARY, "light_front_second")
-ADD_ENUM(LIGHT_REAR_SECONDARY, "light_rear_second")
-ADD_ENUM(SEAT_FRONT, "seat_front")
-ADD_ENUM(SEAT_REAR, "seat_rear")
-ADD_ENUM(EXHAUST, "exhaust")
-ADD_ENUM(ENGINE, "engine")
-ADD_ENUM(GAS_CAP, "gas_cap")
-ADD_ENUM(TRAILER_ATTACH, "trailer_attach")
-ADD_ENUM(HAND_REST, "hand_rest")
-ADD_ENUM(EXHAUST_SECONDARY, "exhaust_second")
-ADD_ENUM(WING_AIRTRAIL, "wing_airtrail")
-ADD_ENUM(VEH_GUN, "veh_gun")
+IMPLEMENT_ENUM_BEGIN(eVehicleModelDummy::e)
```

Comment:
`eVehicleModelDummy` is a namespace and `e` is the actual enum. 

```
namespace eVehicleModelDummy
{
    enum e
    {
        LIGHT_FRONT_MAIN = 0,
        LIGHT_REAR_MAIN,
        ...
    };
}
```
This an alternative to enum class. If I use enum class here, then I can't use the enum member as array index. This will throw compile-time errors `vecDummies[eVehicleModelDummy::EXHAUST]`.

But with `eVehicleModelDummy::e`, I can do `vecDummies[eVehicleModelDummy::e::EXHAUST]`. As you can see the enum members are not in global scope. Eeh, it's definitely not the best solution, but I can't think of anything else :P

Reply:
What about making the inner enum a regular `enum` rather than `enum class`?
```
namespace eVehicleModelDummy
{
    enum Enum
    {
        LIGHT_FRONT_MAIN = 0,
        LIGHT_REAR_MAIN,
    };
}
```
This gives us proper scoping into `eVehicleModelDummy`, but still allows using it as an array index. Only downside of this would be that we need to specify `eVehicleModelDummy::Enum` as a type in some places.

Reply:
Perhaps just cast when using it as an array index?
eg.:
```cpp
enum class eVehicleModelDummy
{
        LIGHT_FRONT_MAIN = 0,
        LIGHT_REAR_MAIN,
};
// Somewhere in the code...
array[(size_t)eVehicleModelDummy::LIGHT_REAR_MAIN)]
```
You could cast to `int` as well, it doesn't really matter. (In the the compiler will use `size_t` probably..)

Reply:
I think that defeats the purpose of using enum class if we're going to use casts. More importantly, I'd like to use static_cast rather than raw casts, but then it just makes the code look complicated.